sdks/csharp/sdk/AgonesSDK.cs

```diff
@@ -128,34 +127,35 @@ public async Task<Status> ReserveAsync(long seconds)
 		{
 			try
 			{
-				await client.ReserveAsync(new Duration() {
-					Seconds = seconds
-				}, deadline: DateTime.UtcNow.AddSeconds(RequestTimeout), cancellationToken: ctoken);
+				await client.ReserveAsync(new Duration { Seconds = seconds},
+					deadline: DateTime.UtcNow.AddSeconds(RequestTimeout),
+					cancellationToken: ctoken);
 				return new Status(StatusCode.OK, $"Reserve({seconds}) request successful.");
 			}
 			catch (RpcException ex)
 			{
-				Console.Error.WriteLine(ex.Message);
+				LogError(ex, "Unable to mark the GameServer to 'Reserved' state.");
 				return ex.Status;
 			}
 		}
 
 		/// <summary>
-		/// This returns most of the backing GameServer configuration and Status. 
+		/// This returns most of the backing GameServer configuration and Status.
 		/// This can be useful for instances where you may want to know Health check configuration, or the IP and Port the GameServer is currently allocated to.
 		/// </summary>
 		/// <returns>A GameServer object containing this GameServer's configuration data</returns>
 		public async Task<GameServer> GetGameServerAsync()
 		{
 			try
 			{
-				return await client.GetGameServerAsync(new Empty(), deadline: DateTime.UtcNow.AddSeconds(RequestTimeout), cancellationToken: ctoken);
+				return await client.GetGameServerAsync(new Empty(),
+					deadline: DateTime.UtcNow.AddSeconds(RequestTimeout),
+					cancellationToken: ctoken);
 			}
 			catch (RpcException ex)
 			{
-				Console.Error.WriteLine(ex.Message);
-				//Should I rethrow the exception?
-				return null;

```

Comment:
This is change in the behavior. I agree that this is how it should have been handled originally, but may break a customer use-case that is already depending on this behavior.
@markmandel would it be ok to suddenly throw exception GetGameServerAsync()? How other SDKs are handling this?

Reply:
Yeah, I think it's reasonable to throw an exception here. I'm also assuming that in C#, exception throwing is the appropriate way to deal with errorrs?

For example, [Go returns an error](https://github.com/googleforgames/agones/blob/master/sdks/go/sdk.go#L74-L77)

The only maybe interesting part, is - do we want to throw the GRPC exception, or do we want to wrap it in our own exception type?

Reply:
Ah, I forgot about this.  Yeah, I figured surfacing the exception would be better than just sinking it like it was previously doing.  Typically, libraries shouldn't do this and should surface it to the consuming callsite.

As for whether or not to make a custom AgonesException to wrap it in, I'll leave it up to you.  I'm fine with handling a gRPC exception in my code.

Reply:
If there is no dependency to this SDK that this change would break them, it looks good to me. I don't think we need to wrap the exception.